### PR TITLE
fix: disable form submission for isPending

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 
from https://github.com/adobe/react-spectrum/issues/6197

This will disable implicit and explicit form submission for pending buttons of type="submit"

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

### CODE REVIEW

**Comment by snowystinger** [comment]
@mjawhariPT first we need to fix the failing tests, then it'll be reviewed by two people on the team. For more information about our releases, see our docs https://react-spectrum.adobe.com/Support.html#release-schedule

**Review by ktabors — DISMISSED** [review]
The closest I found to testing this in storybook or docs was S2 Button with the submit and isPending prop and seeing the blocker button render.

File: packages/react-aria-components/src/Button.tsx [hunk 1]
```diff
@@ -156,27 +156,47 @@ export const Button = /*#__PURE__*/ createHideableComponent(function Button(prop
...
+    },
+    onClick: (e) => {
+      if (e.currentTarget instanceof HTMLButtonElement) {
+        e.preventDefault();
+      }
+    }
+  } : {}, [isPending]);
 
+  // When the button is in a pending state, we want to stop implicit form submission (ie. when the user presses enter on a text input).
+  // We do this by rendering a hidden submit button that is disabled BEFORE the actual submit button as a form's default button is the first submit button
```
Comment (devongovett) [inline packages/react-aria-components/src/Button.tsx:173]: Wouldn't preventing default in onClick of the regular button handle that? Why do we need a second button?

Comment (devongovett) [reply packages/react-aria-components/src/Button.tsx:173]: Or change the `type="submit"` to `type="button"`?

Comment (snowystinger) [reply packages/react-aria-components/src/Button.tsx:173]: changing the button type might be a better alternative, but preventing default didn't appear to do anything

Comment (snowystinger) [reply packages/react-aria-components/src/Button.tsx:174]: I'm not quite clear on why a single input will still submit implicitly
I could get the owning form and prevent it from there instead possibly?

Comment (devongovett) [reply packages/react-aria-components/src/Button.tsx:174]: so is the onKeyDown and onClick handler needed anymore since you changed the type?

**Review by reidbarber — APPROVED** [review]

**Review by ktabors — APPROVED** [review]

**Pull request merged**